src/NHibernate/Cache/ReadWriteCache.cs

```diff
@@ -157,30 +130,36 @@ public object[] GetMany(CacheKey[] keys, long timestamp)
 			var result = new object[keys.Length];
 			lock (_lockObject)
 			{
-				var lockables = _cache.GetMany(keys.Select(o => (object) o).ToArray());
+				var lockables = _cache.GetMany(keys.ToArray<object>());
```

Comment:
Why not `_cache.GetMany(keys)`? Same question for all other places with `ToArray` conversion

Reply:
Same as above.